nucleus/core/logging/src/main/java/com/sun/enterprise/server/logging/logviewer/backend/LogFilterForInstance.java

```diff
@@ -178,14 +184,60 @@ public File downloadGivenInstanceLogFile(ServiceLocator habitat, Server targetSe
 
                 wrf.copyTo(instanceLogFile);
             } catch (WindowsException ex) {
-                throw new IOException("Unable to download instance log file from DCOM Instance Node");
+                throw new IOException("Unable to download instance log file from DCOM Instance Node", ex);
             }
 
+        } else if (node.getType().equals("DOCKER") || node.getType().equals("TEMP")) {
+            String dockerContainerId = targetServer.getDockerContainerId();
+            String nodeHost = node.getNodeHost();
+            int nodeDockerPort = Integer.valueOf(node.getDockerPort());
+            try {
+                if (dockerContainerId == null && node.getType().equals("TEMP")) {
+                    // FIXME: it would be easies, if the TEMP server had docker container id set!
```

Comment:
An instance on a TEMP node _should_ have it set?

From the docker-node entrypoint script, it should get set when creating the instance:
```
function createNewInstance {
...
    if [ ! -z "${DOCKER_CONTAINER_ID}" ]; then
       # Register Docker container ID to DAS
       echo "Setting Docker Container ID for instance ${PAYARA_INSTANCE_NAME}: ${DOCKER_CONTAINER_ID}"
       ASADMIN_COMMAND="${ASADMIN} -I false -H ${PAYARA_DAS_HOST} -p ${PAYARA_DAS_PORT} -W ${PAYARA_PASSWORD_FILE} _set-docker-container-id --instance ${PAYARA_INSTANCE_NAME} --id ${DOCKER_CONTAINER_ID}"
       echo "${ASADMIN_COMMAND}"
       ${ASADMIN_COMMAND}
    fi
```

The script even makes efforts to ensure that it at least has _something_ set right at the start to avoid failing that if check:

```
DOCKER_CONTAINER_ID="$(cat /proc/self/cgroup | grep :/docker/  | sed s/\\//\\n/g | tail -1)"
```

It also later on makes efforts to check that the container ID matches:

```
        # Check if an instance with this name is actually registered
        echo "Checking if an instance with name ${PAYARA_INSTANCE_NAME} has been registered with the DAS"
        ...

        if [ ! -z "${INSTANCE_EXISTS}" ]; then
            # Check if Docker container ID registered against the instance name is the same
            echo "Found an instance with name ${PAYARA_INSTANCE_NAME} registered to the DAS, checking if registered Docker Container ID matches this container's ID"
            ASADMIN_COMMAND="${ASADMIN} -I false -t -H ${PAYARA_DAS_HOST} -p ${PAYARA_DAS_PORT} -W ${PAYARA_PASSWORD_FILE} _get-docker-container-id --instance ${PAYARA_INSTANCE_NAME}"
            echo "${ASADMIN_COMMAND}"
            REGISTERED_DOCKER_CONTAINER_ID="$(${ASADMIN_COMMAND})" || true

            if [ ! -z "${REGISTERED_DOCKER_CONTAINER_ID}" ]; then
                # If they're the same, simply create the folders, otherwise create and register a new instance
                echo "Registered Docker Container ID is: ${REGISTERED_DOCKER_CONTAINER_ID}"
                if [ "${REGISTERED_DOCKER_CONTAINER_ID}" == "${DOCKER_CONTAINER_ID}" ]; then
                    echo "Docker Container IDs match, creating local instance filesystem: "
                    ASADMIN_COMMAND="${ASADMIN} -I false -T -H ${PAYARA_DAS_HOST} -p ${PAYARA_DAS_PORT} -W ${PAYARA_PASSWORD_FILE} _create-instance-filesystem --node ${PAYARA_NODE_NAME} --dockernode true ${PAYARA_INSTANCE_NAME}"
                    ${ASADMIN_COMMAND}
                else
                    echo "Docker Container IDs do not match, creating a new instance."
                    createNewInstance
                fi
            else
                echo "Could not retrieve registered Docker Container ID, creating a new instance"
                createNewInstance
            fi
        else
            createNewInstance
        fi
```

So I think the only way it couldn't be set is if you're not using the script? In which case I'd just exit out and reach for my "stop holding it wrong" bat